src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -136,10 +140,55 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
   const selectedDataViewRef = useRef<DataView | null | undefined>(selectedDataView);
   const showFieldList = sidebarState.status !== DiscoverSidebarReducerStatus.INITIAL;
 
+  const querySubscriberResult = useQuerySubscriber({ data });
+
+  const fetchFields = useCallback(async () => {
+    console.log('*** fetchFields is firing');
+    const existingFieldList = await loadFieldExisting({
+      data,
+      dataView: sidebarState.dataView!,
+      fromDate: querySubscriberResult.fromDate || '',
+      toDate: querySubscriberResult.toDate || '',
+      // dslQuery: querySubscriberResult.query || {},
+      dslQuery: await buildSafeEsQuery(
+        sidebarState.dataView!,
+        querySubscriberResult.query!,
+        querySubscriberResult.filters || [],
+        getEsQueryConfig(core.uiSettings)
+      ),
+      timeFieldName: sidebarState.dataView?.timeFieldName,
+      dataViewsService: dataViews,
+      uiSettingsClient: core.uiSettings,
+    });
+
+    dispatchSidebarStateAction({
+      type: DiscoverSidebarReducerActionType.DOCUMENTS_LOADED,
+      payload: {
+        dataView: selectedDataViewRef.current,
```

Comment:
I'm pretty sure I want to use `selectedDataViewRef.current` for all data view refs BUT when I look at the current code, I get a different value before and after `loadFieldExisting` is fired